xilem_core/src/views/memoize.rs

```diff
@@ -42,17 +42,18 @@ It's not possible in Rust currently to check whether the (content of the) callba
     }
 }
 
-impl<State, Action, D, V, F> MasonryView<State, Action> for Memoize<D, F>
+impl<State, Action, Context, Data, V, ViewFn> View<State, Action, Context> for Memoize<Data, ViewFn>
 where
-    D: PartialEq + Send + Sync + 'static,
-    V: MasonryView<State, Action>,
-    F: Fn(&D) -> V + Send + Sync + 'static,
+    Context: ViewPathTracker,
+    Data: PartialEq + 'static,
+    V: View<State, Action, Context>,
+    ViewFn: Fn(&Data) -> V + 'static,
 {
-    type ViewState = MemoizeState<State, Action, V>;
+    type ViewState = MemoizeState<State, Action, Context, V>;
```

Comment:
I notice that `ViewState` is either `()` or some variant of `Child::ViewState` for every single implementation except this one. You might want to add a doc comment in the `View` and `ViewSequence` trait indicating that `Memoize` is the "bottom turtle" of both `ViewState` and `SeqState`.

Reply:
I'm not sure that these docs would be true - the main use case for `ViewState` is to store an inner `SeqState`; the main use case for `SeqState` is actually to store generations to allow proper async wiring.

This is a use case of `ViewState`, but I'm personally inclined to try and find alternative patterns of memoization, because of the hot-reloading use cases.
Although actually doing some more thinking on that; I suppose that the Arc case is actually pretty similar...

Reply:
Yeah, I probably missed some stuff in the rebuild code, then. In any case, I'd like documentation to include a better description of what `ViewState` and `SeqState` do, because on a cursory read the only thing you see is "storing someone else's ViewState/SeqState, which ends up being `()`".